nutanix/resource_nutanix_ndb_register_database.go

```diff
@@ -0,0 +1,697 @@
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"database_cluster_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"database_status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dbserver_logical_cluster_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"time_machine_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"parent_time_machine_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"time_zone": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"info": dataSourceEraDatabaseInfo(),
+			"group_info": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"metadata": dataSourceEraDBInstanceMetadata(),
+			"metric": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"parent_database_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"parent_source_database_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"lcm_config": dataSourceEraLCMConfig(),
+			"dbserver_logical_cluster": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"database_nodes":   dataSourceEraDatabaseNodes(),
+			"linked_databases": dataSourceEraLinkedDatabases(),
+		},
+	}
+}
+
+func resourceNutanixNDBRegisterDatabaseCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*Client).Era
+
+	log.Println("Creating the request!!!")
+	req, err := buildReisterDBRequest(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resp, er := conn.Service.RegisterDatabase(ctx, req)
+	if er != nil {
+		return diag.FromErr(er)
+	}
+	log.Println(resp)
+	d.SetId(resp.Entityid)
+
+	// Get Operation ID from response of RegisterDatabaseResponse and poll for the operation to get completed.
+	opID := resp.Operationid
+	if opID == "" {
+		return diag.Errorf("error: operation ID is an empty string")
+	}
+	opReq := era.GetOperationRequest{
+		OperationID: opID,
+	}
+
+	log.Printf("polling for operation with id: %s\n", opID)
+
+	// Poll for operation here - Operation GET Call
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"PENDING"},
+		Target:  []string{"COMPLETED", "FAILED"},
+		Refresh: eraRefresh(ctx, conn, opReq),
+		Timeout: d.Timeout(schema.TimeoutCreate),
+		Delay:   eraDelay,
+	}
+
+	if _, errWaitTask := stateConf.WaitForStateContext(ctx); errWaitTask != nil {
+		return diag.Errorf("error waiting for db register	 (%s) to create: %s", resp.Entityid, errWaitTask)
+	}
+	return nil
+}
+
+func resourceNutanixNDBRegisterDatabaseRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
Database read can be used from database resource or dataasources.

Reply:
yes. now we are reading from database resource.